nemo/collections/nlp/data/text_classification/text_classification_dataset.py

```diff
@@ -60,193 +60,215 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        sent_lengths = []
+        too_long_count = 0
+        for sent_id, sent in enumerate(all_sents):
+            sent_subtokens = [tokenizer.cls_token]
+            for word in sent:
+                word_tokens = tokenizer.text_to_tokens(word)
+                sent_subtokens.extend(word_tokens)
+
+            if max_seq_length > 0 and len(sent_subtokens) + 1 > max_seq_length:
+                sent_subtokens = sent_subtokens[:max_seq_length]
+                too_long_count += 1
+
+            sent_subtokens.append(tokenizer.sep_token)
+            sent_lengths.append(len(sent_subtokens))
 
             input_ids = [tokenizer.tokens_to_ids(t) for t in sent_subtokens]
 
             # The mask has 1 for real tokens and 0 for padding tokens.
             # Only real tokens are attended to.
             input_mask = [1] * len(input_ids)
+            segment_ids = [0] * len(input_ids)
 
-            # Zero-pad up to the sequence length.
-            while len(input_ids) < max_seq_length:
-                input_ids.append(0)
-                input_mask.append(0)
-            segment_ids = [0] * max_seq_length
-
-            assert len(input_ids) == max_seq_length
-            assert len(input_mask) == max_seq_length
-
-            if sent_id < 5:
+            if verbose and sent_id < 2:
                 logging.info("*** Example ***")
-                logging.info("example_index: %s" % sent_id)
+                logging.info(f"example {sent_id}: {sent}")
                 logging.info("subtokens: %s" % " ".join(sent_subtokens))
-                logging.info("label: %s" % label)
                 logging.info("input_ids: %s" % list2str(input_ids))
+                logging.info("segment_ids: %s" % list2str(segment_ids))
                 logging.info("input_mask: %s" % list2str(input_mask))
+                logging.info("label: %s" % labels[sent_id] if labels else "**Not Provided**")
+
+            label = labels[sent_id] if labels else -1
+            features.append([np.asarray(input_ids), np.asarray(segment_ids), np.asarray(input_mask), label])
 
-            self.features.append(
-                InputFeatures(
-                    sent_id=sent_id, label=label, input_ids=input_ids, input_mask=input_mask, segment_ids=segment_ids,
-                )
+        if max_seq_length > -1 and too_long_count > 0:
+            logging.warning(
+                f'Found {too_long_count} out of {len(all_sents)} sentences with more than {max_seq_length} subtokens. '
+                f'Truncated long sentences from the end.'
             )
+        if verbose:
+            get_stats(sent_lengths)
+        return features
+
+    @staticmethod
+    def cache_features(cached_features_file, features):
+        with open(cached_features_file, 'wb') as f:
+            pickle.dump(features, f, protocol=pickle.HIGHEST_PROTOCOL)
+
+    @staticmethod
+    def load_cached_features(cached_features_file):
+        with open(cached_features_file, "rb") as input_file:
```

Comment:
dosctring+ types

Reply:
Is that docstring mandatory for functions called by user directly?